tests/test_aliases.py

```diff
@@ -178,6 +198,10 @@ class Child(Parent):
 
 @pytest.mark.xfail(reason='working on V2')
 def test_low_priority_alias():
+    # TODO: alias_priority has been removed from `Field` — what should happen to this test?
+    #   Should we re-add alias_priority as a keyword argument to `Field`?
+    #   Is there something new that can be used to replicate this functionality?
+    #   Either way, if we don't re-add it to `Field`, remember to update the migration guide
```

Comment:
Question 3: Should we re-add `alias_priority` to `Field`? I am inclined to say yes for backwards compatibility's sake since it doesn't seem to have much impact, but I don't have to if we want to remove it.

Reply:
In this case, it provides a way to ensure that a field alias gets changed by an alias generator.

Since we are getting rid of the ability to change aliases of fields defined in parent classes using `Config.fields`, we should be thoughtful about whether we choose to support this, because it may result in people using `alias_generator` as a way to retain their ability to specify aliases in subclasses.

I see arguments for and against:
* For: this is a migration path for overriding parent class aliases, even if it ends up deprecated
* Against: alias_generator is treated as opaque by the mypy plugin (that said, I'm not sure if that's also the case for `Config.fields`, but I suspect the mypy plugin does inspect its values). Because of that, supporting this as a viable alternative might push people in a direction where they end up frustrated by this as a regression.

If there was any way to "redefine" a field on a child class without emitting a warning about shadowing I think that would address this as an issue, but I'd also be open to just telling people "you can't override aliases in submodels", and tell them if they want different aliases at different points in the class hierarchy they need to create parent models without aliases and subclass them with alias generators (not that that's a particularly elegant solution, but I think it would work, modulo inheritance wonkiness that's probably best to avoid anyway).